homeassistant/components/ssdp/__init__.py

```diff
@@ -85,6 +88,65 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class _HaServiceDescription:
+    """Keys added by HA."""
+
+    x_homeassistant_matching_domains: set[str]
+
+
+@dataclass
+class _SsdpServiceDescription:
+    """SSDP info with optional keys."""
+
+    ssdp_usn: str
+    ssdp_st: str
+    ssdp_location: str | None = None
```

Comment:
Can location actually be None?

Reply:
I'm pretty sure that is what @chishm recommended, but I can't find the comment/review anymore.
@chishm can you confirm?

Reply:
If it can be `None` we should reject it there and make it required.  Its more likely to break internals than be useful if its missing